pkg/controller/logstash/logstash_controller_test.go

```diff
@@ -9,31 +9,55 @@ import (
 	"reflect"
 	"testing"
 
+	"k8s.io/apimachinery/pkg/util/uuid"
+
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/tools/record"
-
+	"k8s.io/utils/pointer"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	commonv1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/common/v1"
 	logstashv1alpha1 "github.com/elastic/cloud-on-k8s/v2/pkg/apis/logstash/v1alpha1"
 	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common"
 	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/comparison"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/expectations"
 	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/hash"
 	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/common/watches"
+	"github.com/elastic/cloud-on-k8s/v2/pkg/controller/logstash/labels"
 	"github.com/elastic/cloud-on-k8s/v2/pkg/utils/k8s"
 )
 
+var (
+	truePtr            = true
+	sampleStorageClass = storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{Name: "fixed"},
+	}
+	fixedStorageClass = storagev1.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{Name: "fixed"},
+	}
+	resizableStorageClass = storagev1.StorageClass{
+		ObjectMeta:           metav1.ObjectMeta{Name: "resizable"},
+		AllowVolumeExpansion: &truePtr,
+	}
+)
+
 func newReconcileLogstash(objs ...client.Object) *ReconcileLogstash {
+	client := k8s.NewFakeClient(objs...)
+
 	r := &ReconcileLogstash{
-		Client:         k8s.NewFakeClient(objs...),
+		Client:         client,
 		recorder:       record.NewFakeRecorder(100),
 		dynamicWatches: watches.NewDynamicWatches(),
+		expectations:   expectations.NewExpectations(client),
```

Comment:
I am wondering if we need an expectation per NamespacedName similar to [es](https://github.com/elastic/cloud-on-k8s/blob/main/pkg/controller/elasticsearch/elasticsearch_controller.go#L76). What would happen if multiple logstash clusters share the same expectation?